File: backend/app/schemas/stats.py
"""
Statistics schemas
"""
from pydantic import BaseModel
from typing import List, Dict, Any
from datetime import datetime


class OverviewStats(BaseModel):
    """Overview statistics for dashboard"""
    new_messages_count: int
    pending_drafts_count: int
    actionables_today_count: int
    total_messages: int
    connected_platforms: List[str]


class ActionableStats(BaseModel):
    """Statistics about actionable items"""
    total_actionables: int
    completed_count: int
    pending_count: int
    overdue_count: int
    by_type: Dict[str, int]
    upcoming_deadlines: List[Dict[str, Any]]


class PlatformStats(BaseModel):
    """Statistics for a specific platform"""
    platform: str
    message_count: int
    last_sync: datetime
    avg_priority: float


class SpamStats(BaseModel):
    """Statistics about spam detection"""
    total_spam_count: int
    spam_by_type: Dict[str, int]
    spam_percentage: float
    recent_spam: List[Dict[str, Any]]


class PriorityDistribution(BaseModel):
    """Distribution of messages by priority"""
    high_priority: int  # 0.7-1.0
    medium_priority: int  # 0.4-0.6
    low_priority: int  # 0.0-0.3
    avg_priority: float

File: frontend/src/services/stats.ts
import api from './api';

export interface OverviewStats {
    new_messages_count: number;
    pending_drafts_count: number;
    actionables_today_count: number;
    total_messages: number;
    connected_platforms: string[];
}

export interface SpamStats {
    total_spam_count: number;
    spam_by_type: Record<string, number>;
    spam_percentage: number;
    recent_spam: Array<{
        message_id: string;
        sender: string;
        subject: string;
        spam_type: string;
        spam_score: number;
        unsubscribe_link: string;
        timestamp: string;
    }>;
}

export interface PriorityDistribution {
    high_priority: number;
    medium_priority: number;
    low_priority: number;
    avg_priority: number;
}

export interface ActionableStats {
    total_actionables: number;
    completed_count: number;
    pending_count: number;
    overdue_count: number;
    by_type: Record<string, number>;
    upcoming_deadlines: Array<{
        id: string;
        description: string;
        type: string;
        deadline: string;
        message_id: string;
    }>;
}

export const statsService = {
    async getOverviewStats(): Promise<OverviewStats> {
        const response = await api.get('/api/stats/overview');
        return response.data;
    },

    async getSpamStats(): Promise<SpamStats> {
        const response = await api.get('/api/stats/spam');
        return response.data;
    },

    async getPriorityDistribution(): Promise<PriorityDistribution> {
        const response = await api.get('/api/stats/priority-distribution');
        return response.data;
    },

    async getActionableStats(): Promise<ActionableStats> {
        const response = await api.get('/api/stats/actionables');
        return response.data;
    }
};
